workshops/models.py

```diff
@@ -227,9 +228,14 @@ def ongoing_events(self):
         return queryset
 
     def unpublished_events(self):
-        '''Return a queryset for events that are not yet published.'''
+        '''Return events without URLs that are upcoming or have unknown starts.
 
-        return self.filter(published=False)
+        Events are ordered by slug and then by serial number.'''
+
+        future_without_url = (Q(start__gte=datetime.date.today()) & Q(url__isnull=True))
```

Comment:
No  need for the wrapping parens unless you're using them for line continuation.